src/coreclr/jit/fgopt.cpp

```diff
@@ -537,7 +537,8 @@ PhaseStatus Compiler::fgPostImportationCleanup()
                 //
                 BasicBlock* const oldTryEntry  = HBtab->ebdTryBeg;
                 BasicBlock*       tryEntryPrev = oldTryEntry->Prev();
-                while ((tryEntryPrev != nullptr) && tryEntryPrev->HasFlag(BBF_REMOVED))
+                assert(tryEntryPrev != nullptr);
+                while (tryEntryPrev->HasFlag(BBF_REMOVED))
                 {
                     tryEntryPrev = tryEntryPrev->Prev();
```

Comment:
Add another assert in here at the end?

Reply:
(or rather, remove the assert below into the loop)

Reply:
there is already a trailing assert:

![image](https://github.com/dotnet/runtime/assets/523221/0c95702f-8bfc-4326-81eb-d88a94677631)

Reply:
moved it into the loop